include/circt/Translation/ExportVerilog.h

```diff
@@ -27,7 +30,37 @@ class ModuleOp;
 
 namespace circt {
 
-/// Export a module containing RTL, and SV dialect code.
+/// Options which control the emission from CIRCT to Verilog.
+struct VerilogEmitterOptions {
+  /// Error callback type used to indicate errors parsing the options string.
+  using ErrorHandlerT = function_ref<void(llvm::Twine)>;
+
+  /// Create a VerilogEmitterOptions with the default values.
+  VerilogEmitterOptions() = default;
+
+  /// Create a VerilogEmitterOptions with the default values overriden by the
+  /// options string.
+  VerilogEmitterOptions(StringRef options, ErrorHandlerT callback)
+      : VerilogEmitterOptions() {
+    parse(options, callback);
+  }
+
+  /// Read in options from a string, overriding only the set options in the
+  /// string.
+  void parse(StringRef options, ErrorHandlerT callback);
+
+  /// Returns a string representation of the options.
+  std::string toString() const;
+
+  // Print the serialized form to an output stream.
+  void print(llvm::raw_ostream &ostream) const;
```

Comment:
I'd recommend dropping the print method, it is trivial sugar.  If this is important, it should be spelled `os << yourOptions` instead of `yourOptions.print(os)`.

Reply:
dropped!